executor/slow_query.go

```diff
@@ -118,16 +116,49 @@ func (e *slowQueryRetriever) close() error {
 	return nil
 }
 
-func (e *slowQueryRetriever) dataForSlowLog(ctx sessionctx.Context) ([][]types.Datum, error) {
-	reader := bufio.NewReader(e.files[e.fileIdx].file)
-	rows, err := e.parseSlowLog(ctx, reader, 1024)
+type parsedSlowLog struct {
+	rows [][]types.Datum
+	err  error
+}
+
+func (e *slowQueryRetriever) parseDataForSlowLog(ctx context.Context, sctx sessionctx.Context) {
+	reader := bufio.NewReader(nil)
+	for e.fileIdx < len(e.files) {
+		reader.Reset(e.files[e.fileIdx].file)
```

Comment:
The `bufio.Reader` should be reset only when the underlying file changed.

Reply:
Thanks for your suggestion!